src/app/leaves/add/components/LeaveRequestFormBuilder.tsx

```diff
@@ -0,0 +1,271 @@
+      status: "Pending",
+    },
+    mode: "all",
+  });
+
+  const { errors, isValid } = form.formState;
+
+  function onSubmit(values: z.infer<typeof leaveRequestSchema>) {
+    console.log(values);
+  }
+  return (
+    <FormWrapper>
+      <h1 className="text-sm text-sm font-semibold text-gray-800 my-4">
+        Request Leave
+      </h1>
+      <Form {...form}>
+        <form
+          data-testid="form"
+          onSubmit={form.handleSubmit(onSubmit)}
+          className="space-y-4"
+        >
+          <MemberSelector control={form.control} name="memberId" />
+
+          <FormField
+            control={form.control}
+            name="leaveType"
+            render={({ field }) => (
+              <FormItem>
+                <FormLabel>Leave Type</FormLabel>
+                <FormControl>
+                  <Select
+                    {...field}
+                    value={field.value}
+                    onValueChange={field.onChange}
+                  >
+                    <SelectTrigger data-testid="leave-type" id="leaveType">
+                      <SelectValue placeholder="Select a leave type" />
+                    </SelectTrigger>
+                    <SelectContent>
+                      {leaveRequestType.map((leaveTye) => (
+                        <SelectItem key={leaveTye.leaveTypes} value={leaveTye.leaveTypes}>
+                          {leaveTye.leaveTypes}
+                        </SelectItem>
+                      ))}
+                    </SelectContent>
+                  </Select>
+                </FormControl>
+                <FormMessage />
+              </FormItem>
+            )}
+          />
+          <FormField
+            control={form.control}
+            name="startDate"
+            render={({ field }) => (
+              <FormItem className="flex flex-col">
+                <FormLabel>Start Date</FormLabel>
+                <Popover>
+                  <PopoverTrigger asChild>
+                    <FormControl>
+                      <Button
+                        variant={"outline"}
+                        data-testid="calendarButton"
+                        className={cn(
+                          `pl-3 text-left font-normal ${
+                            errors.startDate
+                              ? "border-destructive"
+                              : "border-input"
+                          }`
+                        )}
+                      >
+                        {field.value ? (
+                          format(field.value, "PPP")
+                        ) : (
+                          <span className="text-xs">Pick a date</span>
+                        )}
+                        <CalendarIcon className="ml-auto h-4 w-4 opacity-50" />
+                      </Button>
+                    </FormControl>
+                  </PopoverTrigger>
+                  <PopoverContent className="w-auto p-0" align="start">
+                    <Calendar
```

Comment:
Let's use @Mubashar1009 component. It's more sophisticated. Maybe daterange can replace it.